apps/front/src/components/ui/formInput/formInput.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    IconButton,
+    InputAdornment,
+    TextField,
+    TextFieldProps,
+} from '@mui/material'
+import React, { FC } from 'react'
+import { Controller, get, useFormContext } from 'react-hook-form'
+import { Visibility, VisibilityOff } from '@mui/icons-material'
+import useFormInput from '@src/components/ui/formInput/useFormInput'
+
+type FormInputProps = {
+    name: string
+    isPassword?: boolean
+} & TextFieldProps
+
+const FormInput: FC<FormInputProps> = ({ name, isPassword, ...otherProps }) => {
+    const {
+        control,
+        formState: { errors },
+        trigger,
+        setValue,
+    } = useFormContext()
```

Comment:
te wszystkie rzeczy są dostępne z poziomu `Controller` więc możesz masz tu miejsce na pozbycie się tego hook'a.

Reply:
Czy możesz trochę lepiej wytłumaczyć jak mogę to zrobić? Zrobiłem research i nie wiem jak np dostać się do trigger z poziomu Controller. formState: { errors } używam w: ` const error = get(errors, name) ` tak jak w todolist. Może nie rozumiem w pełni polecenia

Reply:
gdzie zrobiłeś research? https://react-hook-form.com/docs/usecontroller/controller masz podany formState i nawet fieldState, który zawiera w sobie `error` z konkretnego pola. NIe trzeba wtedy używać tego `get(errors, name)`